packages/api-docs-builder/ApiBuilders/ComponentApiBuilder.ts

```diff
@@ -534,21 +534,20 @@ const generateComponentApi = async (componentInfo: ComponentInfo, project: TypeS
           astTypes.visit(ast, {
             visitVariableDeclaration: (variablePath) => {
               const definitions: any[] = [];
-              if (variablePath.node.declarations) {
-                variablePath
-                  .get('declarations')
-                  .each((declarator: any) => definitions.push(declarator.get('init')));
-              }
-
-              definitions.forEach((definition) => {
-                if (definition.value?.callee) {
-                  const definitionName = definition.value.callee.name;
-
-                  if (definitionName === `create${name}`) {
-                    node = definition;
-                  }
+              /* 
+              `variablePath.get('declarations')` and
+              `variablePath.node.declarations` return different data
```

Comment:
Should we then use `variablePath.get('declarations')` both here and in the iteration below? What exactlly is different between the two structures?

Reply:
```jsx
(declarator.init.callee?.name === `create${name}`)
```

is always `false` if we iterate using items of `variablePath.get('declarations')` rather than `variablePath.node.declarations` as `calle.name` isn't there.

However, this issue seems to occur only in this branch: https://github.com/mui/material-ui/pull/36008

I am trying to figure out why